shell/platform/common/text_input_model_unittests.cc

```diff
@@ -403,6 +403,17 @@ TEST(TextInputModel, UpdateComposingRemovesLastComposingCharacter) {
   model->SetText("ACDE");
 }
 
+TEST(TextInputModel, UpdateSelectionWhileComposing) {
+  auto model = std::make_unique<TextInputModel>();
+  model->SetText("ABCDE");
+  model->BeginComposing();
+  model->SetComposingRange(TextRange(4, 5), 1);
+  model->UpdateComposingText(u"ぴょんぴょん", TextRange(3, 6));
```

Comment:
nit: this defintely needs a 🐇 emoji in it.

Reply:
I tried but that emoji has 4 utf-8 code units in it, so it may not be the easiest for future readers to figure out what is going on.

Reply:
It was worth a try!